addons/account/demo/account_demo.py

```diff
@@ -0,0 +1,352 @@
+            }),
+            (f'{cid}_invoice_activity_3', {
+                'res_id': ref(f'account.{cid}_demo_invoice_1').id,
+                'res_model_id': ref('account.model_account_move').id,
+                'activity_type_id': ref('mail.mail_activity_data_todo').id,
+                'date_deadline': (fields.Datetime.today() + relativedelta(days=5)).strftime('%Y-%m-%d %H:%M'),
+                'summary': 'Include upsell',
+                'create_uid': ref('base.user_admin').id,
+                'user_id': ref('base.user_admin').id,
+            }),
+            (f'{cid}_invoice_activity_4', {
+                'res_id': ref(f'account.{cid}_demo_invoice_extract').id,
+                'res_model_id': ref('account.model_account_move').id,
+                'activity_type_id': ref('mail.mail_activity_data_todo').id,
+                'date_deadline': (fields.Datetime.today() + relativedelta(days=5)).strftime('%Y-%m-%d %H:%M'),
+                'summary': 'Update address',
+                'create_uid': ref('base.user_admin').id,
+                'user_id': ref('base.user_admin').id,
+            }),
+        ])
+
+    @api.model
+    def _post_create_demo_data(self, created):
+        cid = self.env.company.id
+        if created._name == 'account.move':
+            created = created.with_context(check_move_validity=False)
+            # We need to recompute some oncganges. Invoice lines VS journal items are already done
+            # synchorinzed in the create, but the onchange were not applied on the invoice lines.
+            for move in created:
+                move._onchange_partner_id()
+
+            created.line_ids._onchange_product_id()
+            created.line_ids._onchange_account_id()
+
+            created._recompute_dynamic_lines(
+                recompute_all_taxes=True,
+                recompute_tax_base_amount=True,
+            )
+
+            # the invoice_extract acts like a placeholder for the OCR to be ran and doesn't contain
+            # any lines yet
+            for move in created - self.env.ref(f'account.{cid}_demo_invoice_extract'):
+                try:
+                    move.action_post()
+                except (UserError, ValidationError):
+                    _logger.exception('Error while posting demo data')
+        elif created._name == 'account.bank.statement':
+            created.button_post()
+
+    @api.model
+    def _get_demo_account(self, xml_id, user_type_id, company):
+        """Find the most appropriate account possible for demo data creation.
+
+        :param xml_id (str): the xml_id of the account template in the generic coa
+        :param user_type_id (str): the full xml_id of the account type wanted
+        :param company (Model<res.company>): the company for which we search the account
+        :return (Model<account.account>): the most appropriate record found
+        """
+        return (
+            self.env['account.account'].browse(self.env['ir.model.data'].search([
```

Comment:
@rco-odoo the Many2oneReference fields don't act as recordsets?

Reply:
I guess this answers the question?
```python
In [1]: type(self.env['ir.model.data'].search([('model', '=', 'account.account')], limit=1).res_id)              
Out[1]: int
```